tests/mock_datasource/property_definition.tfmock.hcl

```diff
@@ -0,0 +1,13 @@
+mock_data "opslevel_property_definition" {
+  defaults = {
+    allowed_in_config_files = true
+    description             = "mock-property-definition-description"
+    # id intentionally omitted - will be assigned a random string
+    name                    = "mock-property-definition-name"
+    property_display_status = "visible"
+    # schema value result of running
+    # 'jsonencode({ "$ref" : "#/$defs/MyProp", "$defs" : { "MyProp" : { "properties" : { "name" : { "type" : "string", "title" : "the new name", "description" : "The name of a friend", "default" : "alex", "examples" : ["joe", "lucy"] } }, "additionalProperties" : false, "type" : "object", "required" : ["name"] } } })'
+    schema = "{\"$defs\":{\"MyProp\":{\"additionalProperties\":false,\"properties\":{\"name\":{\"default\":\"alex\",\"description\":\"The name of a friend\",\"examples\":[\"joe\",\"lucy\"],\"title\":\"the new name\",\"type\":\"string\"}},\"required\":[\"name\"],\"type\":\"object\"}},\"$ref\":\"#/$defs/MyProp\"}"
```

Comment:
Nope, I tried that 😁 
The capabilities in [Terraform test](https://developer.hashicorp.com/terraform/language/tests) are something of a subset of what we get with "normal" terraform configs and `terraform console`

Reply:
Running `cd tests ; terraform test` returns:
```terraform
╷
│ Error: Function calls not allowed
│ 
│   on mock_datasource/property_definition.tfmock.hcl line 10, in mock_data "opslevel_property_definition":
│   10:     schema = jsonencode({ "$ref" : "#/$defs/MyProp", "$defs" : { "MyProp" : { "properties" : { "name" : { "type" : "string", "title" : "the new name", "description" : "The name of a friend", "default" : "alex", "examples" : ["joe", "lucy"] } }, "additionalProperties" : 
false, "type" : "object", "required" : ["name"] } } })
│ 
│ Functions may not be called here.
```

It seems that Terraform limits what `.tfmock.hcl` files can do since these are viewed as data only, and `tftest.hcl` files have more capabilities since we're testing everything terraform would normally handle.